File: Java/Stacks/src/ThirdExerciseProgram.java
import java.lang.reflect.Array;

public class ThirdExerciseProgram {

    public Boolean exec(String word) throws IllegalArgumentException {
        char[] subject;
        int counter = 0;
        int midRange;
        int high;

        if (word != null) {
            subject = word.toLowerCase().toCharArray();
            midRange = word.length() / 2;
            high = word.length() - 1;
        } else {
            throw new IllegalArgumentException("Null value is not allowed");
        }

        while (counter < midRange) {
            if (subject[counter] == subject[high]) {
                counter++;
                high--;
            } else {
                break;
            }
        }

        if (counter == midRange) {
            return true;
        } else {
            return false;
        }
    }
}

File: Kotlin/Stacks/src/ThirdExerciseProgram.kt
class ThirdExerciseProgram {

    fun exec(word: String): Boolean {
        val midRange: Int = word.length / 2
        var high: Int = word.length - 1
        var counter: Int = 0

        while (counter < midRange) {
            if (word[counter].lowercase().trim() == word[high].lowercase().trim()) {
                counter++
                high--
            } else {
                break
            }
        }

        if (counter == midRange) {
            return true
        }else{
            return false
        }

    }
}